### ISSUE TITLE
Render text style only once for a multi geometry

### ISSUE THREAD

**Issue by yaugenka** [openlayers/openlayers#7052]
Currently if you define a text style for e.g. a multipolygon geometry, the text will be rendered as many time as there are polygons in that geometry. See this screenshot for an example
![untitled](https://user-images.githubusercontent.com/14937001/28645562-b66bfb70-7266-11e7-9d39-b7a0da8eb84c.png)

It would be nice to have some kind of an option in the text style which would limit the rendeing to only once for the whole geometry.

**Comment by ahocevar** [comment]
You can easily achieve that with a custom geometry function in your style. The function should decide the polygon to be labeled, and return an interior point for that polygon Something like
```js
// sort geometries descending by extent width, so the widest one will be labeled
function sortByWidth(a, b) {
  return ol.extent.getWidth(b.getExtent()) - ol.extent.getWidth(a.getExtent());
}

new ol.style.Style({
  text: new ol.style.Text({ /* ... */ }),
  geometry: function(feature) {
      var geometry = feature.getGeometry();
      if (geometry.getType() == 'MultiPolygon') {
        var geometries = geometry.getPolygons();
        geometry = geometries.sort(sortByWidth)[0];
      }
      return geometry.getInteriorPoint();
    }
});
```
